[BugFix] Fix load Problem for generated column 2(#25828)

srlch opened the pull request:
Fixes #25828
Problem:
1. In some case, the ref column for the generated column expression maybe cast
   into archer for some purpose. We should cast back into the correct type when
   we construct the plan for generated column in LOAD.
2. Fix the problem that the ref column is appear as placeholder and expression.

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 11 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/Load.java|0|11|00.00%|[811, 813, 814, 815, 816, 818, 823, 824, 825, 826, 828]|

srlch committed:
[BugFix] Fix tow load Problem for generated column 2(#25828)

Problem:
1. In some case, the ref column for the generated column expression maybe cast
   into archer for some purpose. We should cast back into the correct type when
   we construct the plan for generated column in LOAD.
2. Fix the problem that the ref column is appear as placeholder and expression.

Signed-off-by: srlch <linzichao@starrocks.com>

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 11 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/Load.java|0|11|00.00%|[811, 813, 814, 815, 816, 818, 823, 824, 825, 826, 828]|

wanpengfei-git commented:
@Mergifyio backport branch-3.1
